elastic/kibana
# [Fleet] fixing dropping select all

## Summary

Fixes https://github.com/elastic/kibana/issues/133523

Added logic so that query selection (select all) is not dropped on Agent list when data changes.

See detailed reasoning here:
https://github.com/elastic/kibana/issues/133523#issuecomment-1164526372

Didn't modify manual scenario, I think it's not right to keep all 20 agents selected when the underlying data changes, as it might result in taking action on agents that were not intended.

To verify:
- enroll a few agents with horde (more than 20 to have multiple pages)
- select all agents in list
- wait until refresh happens in the background
- expect that the select all (query) state is not lost, and actions can still be taken on all agents
- when unselecting one agent manually, the state goes back to manual selection (e.g. 19 selected instead of 20)

<img width="1006" alt="image" src="https://user-images.githubusercontent.com/90178898/175529079-df621a0a-e14a-47e3-af34-56c601ec6057.png">

After manual unselect (12 agents selected as some of the dropped automatically from view on refresh)
<img width="1006" alt="image" src="https://user-images.githubusercontent.com/90178898/175529186-d2fcc880-1246-4857-a49a-12afdfdf621b.png">



### Checklist
